src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,71 @@
+package core.basesyntax.service;
+
+import core.basesyntax.exceptions.InvalidUserDataException;
+import core.basesyntax.exceptions.NullUserException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final RegistrationServiceImpl registrationService
+            = new RegistrationServiceImpl();
+
+    @Test
+    void register_validUser_ok() {
+        User validUser = new User("Default", "Default", 18);
+        User outputUser;
+        try {
+            outputUser = registrationService.register(validUser);
+            Assertions.assertEquals(validUser, outputUser, "Register returned wrong User");
+        } catch (InvalidUserDataException e) {
+            Assertions.fail("Wrong data validation");
+        }
+    }
+
+    @Test
+    void register_invalidLoginUser_notOk() {
+        User invalidUser = new User("tiny", "Default", 18);
+        Assertions.assertThrows(InvalidUserDataException.class,
+                () -> registrationService.register(invalidUser),
+                "An InvalidUserDataException was expected "
+                        + "for user with login less than 6 characters");
+
+    }
+
+    @Test
+    void register_invalidPasswordUser_notOk() {
+        User invalidUser = new User("Default", "tiny", 18);
+        Assertions.assertThrows(InvalidUserDataException.class,
+                () -> registrationService.register(invalidUser),
+                "An InvalidUserDataException was expected for "
+                + "user with password less than 6 characters");
+
+    }
+
+    @Test
+    void register_invalidAgeUser_notOk() {
+        User invalidUser = new User("Default", "Default", 16);
+        Assertions.assertThrows(InvalidUserDataException.class,
+                () -> registrationService.register(invalidUser),
+                "An InvalidUserDataException was expected for user with age less than 18");
+
+    }
+
+    @Test
+    void register_sameLoginUser_notOk() {
+        User validUser = new User("RomanLoiev", "RomanLoiev", 19);
+        registrationService.register(validUser);
```

Comment:
Ensure that the `register` method correctly handles duplicate logins by checking if the `storageDao` is properly implemented to prevent adding users with the same login.